lib/OpenQA/WebAPI/Controller/File.pm

```diff
@@ -34,7 +34,8 @@ sub needle {
     my ($name, $dummy, $format) = fileparse($self->param('name'), qw(.png .txt));
     my $distri  = $self->param('distri');
     my $version = $self->param('version') || '';
-    my $needle  = OpenQA::Utils::needle_info($name, $distri, $version);
+    my $jsonfile = $self->param('jsonfile') || '';
```

Comment:
since needle_info does not do any location check for `$jsonfile` I would do it here. Else I'm quite afraid of info leak when adversary can add arbitrary path as a `$jsonfile` param